### PR TITLE
TST: Timedelta tests reorg (gh14854)

### PR DESCRIPTION
**Pull request by TrigonaMinima**
- [x] Partially completes #14854 
 - [x] Tests reorganized
 - [x] passes ``git diff upstream/master | flake8 --diff``
 - [x] whatsnew: reorg of Timedelta tests

### CODE REVIEW

File: pandas/tests/indexes/timedeltas/test_to_timedelta.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
+from datetime import time, timedelta
+import numpy as np
+
```
Comment (jreback) [inline pandas/tests/indexes/timedeltas/test_to_timedelta.py]: -> test_tools.py

File: pandas/tseries/tests/test_period.py [hunk 2]
```diff
@@ -6,27 +6,25 @@
 
 """
 
+import numpy as np
+from numpy.random import randn
```
Comment (jreback) [inline pandas/tseries/tests/test_period.py]: you can move this to pandas/tests/scalar/test_period.py

**Comment by jreback** [comment]
@TrigonaMinima looks great!

just a couple of corrections.

File: pandas/tseries/tests/test_period.py (hunk 2 above)
Comment (TrigonaMinima) [inline pandas/tseries/tests/test_period.py]: This is a really big file (5k+ lines)

Still move it there?

**Comment by jreback** [comment]
pls add a whatsnew note as well (reorg of timeseries tests), you can put in other API changes section (in 0.20.0)

**Comment by jreback** [comment]
only thing left after this are the timeseries_legacy.py tests I think (but let's deal with them in a separate PR).

**Comment by TrigonaMinima** [comment]
I haven't finished the reorg of ```Period``` tests yet. Should I do it in this PR only?

File: pandas/tseries/tests/test_period.py (hunk 2 above)
Comment (jreback) [inline pandas/tseries/tests/test_period.py]: sure can do in anthoher PR. I though you moved the PeriodIndex tests out already. But yes, certainly should split it.

**Comment by jreback** [comment]
@TrigonaMinima no, let's leave period for another.

File: pandas/tests/indexes/period/test_period.py [hunk 3]
```diff
@@ -0,0 +1,233 @@
+import numpy as np
+from datetime import timedelta
+
+import pandas as pd
+from pandas.util import testing as tm
```
Comment (jreback) [inline pandas/tests/indexes/period/test_period.py]: so you moved this (just fine).

File: pandas/tests/indexes/test_datetimelike.py [hunk 4]
```diff
@@ -1,467 +0,0 @@
-# -*- coding: utf-8 -*-
-
-import numpy as np
```
Comment (jreback) [inline pandas/tests/indexes/test_datetimelike.py]: and removed all of this (again just fine). that's why I thought you were moving Period. But can certainly finish in another.

**Comment by TrigonaMinima** [comment]
@jreback I tried by best, but there were still some confusions. So, could you please have somewhat deeper look at it. I'll correct the issues if there are any, tomorrow night.

**Comment by jreback** [comment]
what confusions?

**Comment by TrigonaMinima** [comment]
- Some coercion functions from ```tseries/test_tslib.py``` I moved to ```indexing/test_coercion.py```.
- ```_conversion``` and ```_format_converters``` functions (currently in ```scalar/test_timedelta.py```) to be sent to ```scalar/test_timedelta``` or ```indexes/timedeltas```?
- ```test_timedelta_ops_scalar``` (currently in ```indexes/timedeltas/test_ops.py```) in scalar/test_timedelta.py or ```indexes/timedeltas```? Here I was confused because there were some arithmetic operations going on between Timedelta and Timestamp but it said scalar.
- This function - ```test_add_overflow``` was under class ```TestSlicing``` (currently in ```tests/test_ops.py```). Why was that? I almost placed it in ```test_partial_slicing.py```.

**Comment by TrigonaMinima** [comment]
Also, I want to understand the code of the core pandas library. Any suggestions on what should the starting point be?

**Comment by jreback** [comment]
ok i will have a look and see if anything needs moving

so best way to understand code is to:

debug or write a test, then step thru and see what it does

starting with a Series or DataFrame method is useful to see

**Comment by jreback** [comment]
@TrigonaMinima some flake errors.

I'll have a look at your points above when merging.

File: pandas/tests/indexing/test_coercion.py [hunk 5]
```diff
@@ -1260,3 +1263,61 @@ def test_replace_series_timedelta64(self):
 
     def test_replace_series_period(self):
         pass
+
+
+class TestArrayToDatetime(tm.TestCase):
```
Comment (jreback) [inline pandas/tests/indexing/test_coercion.py]: these go in: pandas/tests/datetimes/tests/test_tools.py

**Comment by TrigonaMinima** [comment]
@jreback don't know what to do with this build/merge error.

I have fixed the merge conflicts. I am able to merge this branch to master without any conflicts on my local repo.

**Comment by jreback** [comment]
ok np i'll fix this up

**Comment by jreback** [comment]
https://github.com/pandas-dev/pandas/compare/master...jreback:TrigonaMinima-gh14854-timedelta?expand=1

is the compare, with a couple of changes from on top.

@TrigonaMinima really appreciate the PRs here! segregated tests that are not in huge files are awesome!

**Comment by jreback** [comment]
@TrigonaMinima thanks again for this!

love for you to keep going (period tests!)

keep in mind we are switching to ``pytest`` very shortly (its should work similarly though).

**Pull request closed without merging**